gui/default/syncthing/core/syncthingController.js

```diff
@@ -2227,6 +2258,139 @@ angular.module('syncthing.core')
             }, $scope.emitHTTPError);
         }
 
+        function editFolderLoadingFolderGlobalTreeData() {
+            $scope.folderGlobalTreeData.data = [];
+            $scope.folderGlobalTreeData.fancyData = [];
+            $scope.folderGlobalTreeData.error = null;
+            $scope.folderGlobalTreeData.loading = true;
+        }
+        function editFolderGetFolderGlobalTreeData() {
+            // console.log("editFolderGetFolderGlobalTreeData");
+            return $http.get(urlbase + '/db/browse?folder=' + encodeURIComponent($scope.currentFolder.id))
+                .then(function (r) {
+                    return r.data;
+                }, function (response) {
+                    $scope.folderGlobalTreeData.error = $translate.instant("Failed to load folder global tree.");
+                    return $q.reject(response);
+                });
+        }
+
+        function editFolderLoadFolderGlobalTreeData() {
+            console.log("editFolderLoadFolderGlobalTreeData");
+            editFolderLoadingFolderGlobalTreeData();
+            return editFolderGetFolderGlobalTreeData().then(function (data) {
+                if (!data) {
+                    return;
+                }
+                console.log("editFolderLoadFolderGlobalTreeData | loaded", data.error)
+                setFolderGlobalTreeData(data);
+                $scope.folderGlobalTreeData.error = data.error;
+            }, $scope.emitHTTPError);
+        }
+
+        function setFolderGlobalTreeData(data) {
+            $scope.folderGlobalTreeData.data = data || [];
+
+            // Pre-build $scope.folderGlobalTreeData.data.fancyData
+            // used by fancyTree to speed up fancytree opening later.
+
+            // Worker is preferred if supported so UI don't lag when
+            // parsing tens of thousands files.
+            if (fancyTreeGlobalDataWorker) {
+                fancyTreeGlobalDataWorker.postMessage({
+                    data: $scope.folderGlobalTreeData.data,
+                    ignorePatterns: ignoresArray()
+                });
+            } else {
+                setFolderGlobalTreeFancyData(formatGlobalTreeNodes($scope.folderGlobalTreeData.data, ""))
+            }
+        }
+        function setFolderGlobalTreeFancyData(data) {
+            console.log("setFolderGlobalTreeFancyData", $scope.folderGlobalTreeData.loading, data)
+            $scope.folderGlobalTreeData.fancyData = data;
+            $scope.folderGlobalTree?.reload();
+            $scope.folderGlobalTreeData.loading = false;
+        }
+
+        function formatGlobalTreeNodes(data, parentKey, parentIgnored = true) {
+            var result = [];
+            var ignores = ignoresArray();
+            // console.log("formatGlobalTreeNodes", "data", data, "ignores", ignores);
+            if (ignores[0] !== generatedByTreeSelector) {
+                ignores = [];
+                setIgnoresText([generatedByTreeSelector, "**"]);
```

Comment:
```suggestion
                setIgnoresText([generatedByTreeSelector, "*"]);
```

Very minor, but just one `*` is more than enough. There is no difference between `*` and `**` when used alone like that. There is only a difference when used in actual paths, e.g. `dir/*/file` vs `dir/**/file`.